wfdb/io/record.py

```diff
@@ -2911,59 +2921,91 @@ def wrsamp(
     if "." in record_name:
         raise Exception("Record name must not contain '.'")
     # Check input field combinations
-    if p_signal is not None and d_signal is not None:
+    signal_list = [p_signal, d_signal, e_p_signal, e_d_signal]
+    signals_set = sum(1 for var in signal_list if var is not None)
+    if signals_set != 1:
         raise Exception(
-            "Must only give one of the inputs: p_signal or d_signal"
+            "Must provide one and only one input signal: p_signal, d_signal, e_p_signal, or e_d_signal"
         )
-    if d_signal is not None:
+    if d_signal is not None or e_d_signal is not None:
         if fmt is None or adc_gain is None or baseline is None:
             raise Exception(
-                "When using d_signal, must also specify 'fmt', 'gain', and 'baseline' fields."
+                "When using d_signal or e_d_signal, must also specify 'fmt', 'gain', and 'baseline' fields"
             )
-    # Depending on whether d_signal or p_signal was used, set other
-    # required features.
+
+    # If samps_per_frame is a list, check that it aligns as expected with the channels in the signal
+    if len(samps_per_frame) > 1:
+        # Get properties of the signal being passed
+        non_none_signal = next(signal for signal in signal_list if signal is not None)
+        if isinstance(non_none_signal, np.ndarray):
+            num_sig_channels = non_none_signal.shape[1]
+            channel_samples = [non_none_signal.shape[0]] * non_none_signal.shape[1]
+        elif isinstance(non_none_signal, list):
+            num_sig_channels = len(non_none_signal)
+            channel_samples = [len(channel) for channel in non_none_signal]
+        else:
+            raise TypeError("Unsupported signal format. Must be ndarray or list of lists.")
+
+        # Check that the number of channels matches the number of samps_per_frame entries
+        if num_sig_channels != len(samps_per_frame):
```

Comment:
as above, `len(samps_per_frame)` will error if `samps_per_frame` is an int.

Reply:
Fixed